sim-cli/src/main.rs

```diff
@@ -199,3 +217,47 @@ async fn main() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+async fn read_sim_path(data_dir: PathBuf, sim_file: String) -> anyhow::Result<PathBuf> {
+    let sim_path = data_dir.join(sim_file);
+
+    let sim_path = if sim_path.extension().is_none() {
+        sim_path.with_extension("json")
+    } else {
+        sim_path
+    };
+
+    if sim_path.exists() {
+        Ok(sim_path)
+    } else {
+        let sim_files: Vec<String> = std::fs::read_dir(data_dir.clone())?
+            .filter_map(|f| {
+                f.ok().and_then(|f| {
+                    if f.path().extension()?.to_str()? == "json" {
+                        return f.file_name().into_string().ok();
+                    }
+                    None
+                })
+            })
+            .collect::<Vec<_>>();
+
+        if sim_files.is_empty() {
+            anyhow::bail!("no simulation files found in {:?}.", data_dir);
+        }
+
+        let selection = dialoguer::Select::new()
+            .with_prompt("Select a simulation file")
+            .items(&sim_files)
+            .default(0)
+            .interact()?;
+
+        Ok(data_dir.join(sim_files[selection].clone()))
+    }
```

Comment:
Do we really want to go this route? IMO a file picker for when the user has wrongly selected the file they want to read from is overkill. It's sort of mixing cmd with a GUI

Reply:
Not sure of the mix. This seems like a standard cmd / cli UX, no?
```
$ cargo run -p sim-cli -- -s test
    Finished dev [unoptimized + debuginfo] target(s) in 0.38s
     Running `target/debug/sim-cli -s test`
2023-11-15T19:10:18.036Z INFO  [sim_cli] Simulation file './test.json' does not exist
Select a simulation file:
> sim.json
```
User tries to run simulation with a sim file named `test.json`, but it's not found in the configured/default data directory, so potential alternatives are immediately listed. Seems handy in case of a simple misspelling of the file name.

IMO a UX improvement, though perhaps not asked for by current set of users

Reply:
I just think it is an unnecessary bit of code that, on top of it, adds a dependency. Telling the user the file cannot be found in the location they've specified should be sufficient.

I'm not going to oppose this if there is consensus that this is a nice thing to have, I just find it unnecessary

Reply:
well, I hope its okay to sneak in some nice UX to already happy users :p

Reply:
ack. totally reasonable. See my [comment here](https://github.com/bitcoin-dev-project/sim-ln/pull/132#issuecomment-1813545115)